### DISCUSSION TITLE
SceneViewer - best practice, limitation,...

### DISCUSSION THREAD

**Discussion by JMAR09** [google/model-viewer#3066] [Q&A]
Hi guys, 

during January we should launch 3D/AR feature in our shopping app. It is based on AR Core (SceneViewer) and AR Kit (Quick Look). We would like to understand more where are limitations and we can expect from those solutions. 

Does anybody know any article, instructions, video,... how to use SceneViewer? We are interested with information about limitation (light, dimmension tolerance,...) and simply it would be amazing to find something like SceneViewer tutorial. 

It is quite difficult to find any good source and I am sure that our customers will have many questions later and we have to be prepared for them. The idea is to create short article/video with FAQ about SceneViewer.

Similar problem we have with Apple Quick Look. If you have any idea, where to find more info, it would be great. 

Thank you very much

Looking forward to reading your tips. 

Have a nice day

Jan :)

**Comment by elalish** [comment]
Are you using model-viewer to launch these, or are  you launching Scene  Viewer and Quick Look  directly? If  it's the latter, I don't think I can  do  better than a Google search for their public docs. Neither of the native solutions are highly customizable like WebXR is; their APIs are pretty simple (just a few query params).

**Comment by JMAR09** [comment]
Hi Emmet, thank you for your answer. It looks that I did not describe it clearly. I will try it one more time and I will hope that it will be more clear. 

What I understand that Scene Viewer (SV) and Quick Look (QL) is working in the same way everytime and it does not matter how user download 3D file and opens it. Hope that it is correct. Or am I wrong? I find out that sometimes SV looks different. 

The thing what I would like to solve is to understand why sometimes SV and QL do not work correctly (probably because light, surfaces, camera,...) And on the other hand, what is the ideal enviroment to use SV and QL. 

What I understand from your message - you do not know any official documentation with information what need, right? Or do you have any idea, where I could find any contact for developers of SV and QL? 

Hope now, it is more clear. 

Thank you very much for you time

Have a nice day :) 

Jan



![Scene_viewer](https://user-images.githubusercontent.com/91330743/147107374-a2fe8a5b-aea4-4432-a065-80a3b73bed95.PNG)

![Quick_look](https://user-images.githubusercontent.com/91330743/147107387-6196ad6d-59ef-433c-b628-8482f0882849.PNG)

**Reply by elalish** [reply]
Yeah, that's complicated. AR fundamentally calculates where  planes are (like the floor) by assembling a bunch of roughly coplanar  feature points.  The  points are  located by finding features in  the image (think unique corner-shaped things)  and calculating their parallax over time and  comparing it to the phone's accelerometer. Therefore it won't  work if  there's too  little acceleration, or few feature  points, or those points are not coplanar. Hence the  importance of lighting, floor material, clutter, and motion. If you're having a lot of trouble, putting a few pieces of paper on the  floor can help.
